README.md

```diff
@@ -903,57 +903,82 @@ Homebrew causes insecure directories like `/usr/local/share/zsh/site-functions`
-$ nvm use v14.15.4
-Now using node v14.15.4 (npm v6.14.10)
-```
+Experimental support for the M1 architecture was added in NodeJS v15.3 and full support
+was added in v16.0. Because of this, if you try to install older versions of Node as usual
+you will probably experience either compiliation errors when installing Node or
+out-of-memory errors while running your code.
+
+So, if you want to run a version prior to v16.0 on an M1 Mac, it may be best to compile
+Node targetting the x86_64 Intel architecture so that Rosetta 2 can translate the x86_64
+processor instructions to ARM-based Apple Silicon instructions. Here's what you will need
+to do:
+
+- Install Rosetta, if you haven't already done so
+
+  ```sh
+  $ softwareupdate --install-rosetta
+  ```
+
+  You might wonder, "how will my M1 Mac know to use Rosetta for a version of Node compiled
+  for an Intel chip?". If an executable contains only Intel instructions, macOS will
+  automatically use Rosetta to translate the instructions.
+
+- Open a shell that's running using Rosetta
+
+  ```sh
+  $ arch -x86_64 zsh
+  ```
+
+  Note: This same thing can also be accomplished by finding the Terminal or iTerm App in
+  Finder, right clicking, selecting "Get Info", and then checking the box labeled
+  "Open using Rosetta".
+
+  Note: This terminal session is now running in `zsh`. If `zsh` is not the shell you
+  typically use, `nvm` may not be `source`'d automatically like it probably is for your
+  usual shell through your dotfiles. If that's the case, make sure to source `nvm`.
+
+  ```sh
+  $ source "${NVM_DIR}/.nvm/nvm.sh
+  ```
+
+- Install whatever older version of Node you are interested in. Let's use 12.22.1 as an
+  example. This will fetch the Node source code and compile it, which will take several
+  minutes.
+
+  ```sh
+  $ nvm install v12.22.1 --shared-zlib
+  ```
+
+  Note: You're probably curious why `--shared-zlib` is included. There's a bug in recent
+  versions of Apple's system `clang` compiler. If one of these broken versions is
+  installed on your system, the above step will likely still succeed even if you didn't
+  include the `--shared-zlib` flag. However, later, when you attempt to `npm install`
+  something using your old version of NodeJS, you will see `incorrect data check` errors.
+  If you want to avoid the possible hassle of dealing with this, include that flag. For
+  more details, see [this issue](https://github.com/nodejs/node/issues/39313) and
+  [this comment](https://github.com/nodejs/node/issues/39313#issuecomment-902395576)
+
+- Exit back to your native shell.
+
+  ```sh
+  $ exit
+  $ arch
+  arm64
+  ```
+
+  Note: If you selected the box labeled "Open using Rosetta" rather than running the CLI
+  command in the second step, you will see `i386` here. Unless you have another reason to
+  have that box selected, you can deselect it now.
+
+- Check to make sure the architecture is correct. `x64` is the abbreviation for x86_64,
+  which is what you want to see.
```

Comment:
```suggestion
  Note: If you selected the box labeled "Open using Rosetta" rather than running the CLI command in the second step, you will see `i386` here.
  Unless you have another reason to have that box selected, you can deselect it now.

- Check to make sure the architecture is correct. `x64` is the abbreviation for x86_64, which is what you want to see.
```